sentry_sdk/spotlight.py

```diff
@@ -112,9 +115,18 @@ def setup_spotlight(options):
     else:
         return None
 
-    if settings is not None and env_to_bool(
-        os.environ.get("SENTRY_SPOTLIGHT_ON_ERROR", "1")
+    if (
+        settings is not None
+        and settings.DEBUG
+        and env_to_bool(os.environ.get("SENTRY_SPOTLIGHT_ON_ERROR", "1"))
     ):
-        settings.MIDDLEWARE.append("sentry_sdk.spotlight.SpotlightMiddleware")
+        try:
+            middleware = settings.MIDDLEWARE
+            if DJANGO_SPOTLIGHT_MIDDLEWARE_PATH not in middleware:
+                settings.MIDDLEWARE = type(middleware)(
```

Comment:
i'd always coerce to list instead of this (but can't articulate why, it just "feels less magical" to me)

Reply:
I was gonna do that but then remembered about fiddling with people's settings. I wanna be as hands-off as possible.